src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -272,15 +416,16 @@ def output_dict(self) -> Dict[str, Any]:
 
     def __copy__(self) -> "SnowflakePlan":
         return SnowflakePlan(
-            self.queries.copy() if self.queries else [],
+            copy.deepcopy(self.queries) if self.queries else [],
```

Comment:
Why is deepcopy required here?

Reply:
because self.queries contains Query object, instead of query string, which can be used in original query plan object